crates/bevy_ecs/src/schedule/run_criteria.rs

```diff
@@ -66,7 +68,19 @@ impl From<bool> for ShouldRun {
     }
 }
 
-#[derive(Default)]
+impl Debug for dyn System<In = (), Out = ShouldRun> + 'static {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "System with In=(), Out=ShouldRun")
```

Comment:
We should definitely include the system's name here (and below).

Reply:
Done.